### ISSUE TITLE
Automation API: Tasker, etc.

### ISSUE THREAD

**Issue by artemiv4nov** [AdguardTeam/AdGuardVPNForAndroid#252]
Well, we already have an automation API in the AdGuard application. Our users also want to use this API in AdGuard VPN.
Let's bring it with the following actions:
- VPN turn on/off
- DNS server change

**Comment by DrSuSE** [comment]
Yes, please!  I would love the ability to disconnect/connect via tasker like I do with AdGuard.

**Comment by DrSuSE** [comment]
I was curious if anyone might know of a way to do this even if it's a bit roundabout for right now.

With both Adguard and Adguard VPN I seem to have a bit of trouble when I'm leaving work (Wi-Fi) going to mobile and then getting home (Wi-Fi). The reverse as well.

This is easy to solve with Adguard, tasker restarts protection on network change. I need to do something similar with AVPN.

I do not run them in compatibility mode because I use different things in different situations and I just prefer to run Adjust in root/proxy mode and AVPN in standard VPN mode.

Thank you!

**Comment by TPS** [comment]
This also important d/t Android wanting to kill AGVPN [(as predicted 🤐)](https://github.com/AdguardTeam/AdguardForAndroid/issues/775#issuecomment-863952211), so automation would help ameliorate that, as well.

Also, [AG itself could use better automation to turn on/off AGVPN integration](https://github.com/AdguardTeam/AdguardForAndroid/issues/2673#issuecomment-1871437663).  There's nothing listed for such @ https://adguard.com/kb/adguard-for-android/solving-problems/tasker/

> Let's bring it with the following actions:
> - DNS server change

I'd like to propose a bit of a fine-tuned request here:  I'd like to choose the fastest server by search term (probably country name, but _whatever_, particularly as the server list changes/grows) &/or saved locations.

#306 would easily be resolvable via this.

Especially since v2.8.1n6 (in my setup), the watchdog is effectively useless, so this functionality is now utterly essential.

Maybe https://github.com/AdguardTeam/AdguardForAndroid/issues/4010 could handle _all_ Android AG apps?

> Yes, please!  I would love the ability to disconnect/connect via tasker like I do with AdGuard.

> I was curious if anyone might know of a way to do this even if it's a bit roundabout for right now.

@DrSuSE I've worked around this in some situations by using Tasker & [BuzzKill](https://play.google.com/store/apps/details?id=com.samruston.buzzkill) on AGVPN's <kbd>Connect</kbd> / <kbd>Disconnect</kbd> in the permanent notification.  (I'm reasonably sure it could be done w/ just Tasker; I just find BK's interface simpler.)  Sorry for the late reply!

**Comment by TPS** [comment]
Also, AG itself could use better automation to turn on/off AGVPN integration.  There's nothing listed for such @ https://adguard.com/kb/adguard-for-android/solving-problems/tasker/

**Comment by TPS** [comment]
> Let's bring it with the following actions:
> - DNS server change

I'd like to propose a bit of a fine-tuned request here:  I'd like to choose the fastest server by search term (probably country name, but _whatever_, particularly as the server list changes/grows) &/or saved locations.

**Comment by TPS** [comment]
#306 would be easily resolvable via this.

**Comment by TPS** [comment]
> This also important d/t Android wanting to kill AGVPN [(as predicted 🤐)](https://github.com/AdguardTeam/AdguardForAndroid/issues/775#issuecomment-863952211), so automation would help ameliorate that, as well.

Especially since v2.8.1n6 (in my setup), the watchdog is effectively useless, so this functionality is now utterly essential.

**Comment by TPS** [comment]
Maybe https://github.com/AdguardTeam/AdguardForAndroid/issues/4010 could handle _all_ Android AG apps?

**Comment by TPS** [comment]
> Yes, please!  I would love the ability to disconnect/connect via tasker like I do with AdGuard.

> I was curious if anyone might know of a way to do this even if it's a bit roundabout for right now.

@DrSuSE I've worked around this in some situations by using Tasker & [BuzzKill](https://play.google.com/store/apps/details?id=com.samruston.buzzkill) on AGVPN's <kbd>Connect</kbd> / <kbd>Disconnect</kbd> in the permanent notification.  (I'm reasonably sure it could be done w/ just Tasker; I just find BK's interface simpler.)  Sorry for the late reply!

**Comment by Eloren1** [comment]
bump
